book/testing.md

```diff
@@ -165,4 +165,63 @@ for t in [
 ```
 and be invoked as `nu tests.nu`
 
+### Basic Test Framework
+
+It is also possible to define tests in Nushell as functions with descriptive names and discover
+them dynamically without requiring a [Nupm] package. The following uses `scope commands` and a
+second instance of Nushell to run the generated list of tests.
+
+```nushell
+use std assert
+
+source fib.nu
+
+def main [] {
+  print "Running tests..."
+
+  let test_commands = (
+    scope commands
+      # Exclude all in-built commands
+      | where ($it.type == "custom")
+      # Include all commands starting with "test_"
+      | where ($it.name | str starts-with "test_")
+      # Exclude commands marked as "ignored" in the description
+      | where not ($it.description | str starts-with "ignore")
```

Comment:
Instead of doing `where | where | where` it's typically recommended to use and/or.
```nushell
scope commands | where $it.type == custom and ($it.name | str starts-with test_) and not ($it.description | str starts-with ignore)
```
You version allows for better documentation but I'm not sure it's the most efficient way to write it. I won't block this PR because of this though.

Reply:
That's a fair point. I've got too used to lazily-evaluated sequences perhaps. I've updated based on that. It still seems pretty readable, particularly after removing the line-by-line comments. Plus also added what I intended to push previously (compatibility with nupm tests).